python/lsst/obs/ctio0m9/ingest.py

```diff
@@ -70,18 +72,24 @@ def translate_imgType(self, md):
         if val in conversion:
             return conversion[val]
         else:
+            self.log.warn('Unknown image type %s found in IMAGETYP key', val)
             return 'NOT_TRANSLATED'
 
     def translate_wavelength(self, md):
-        """Get the monochromator wavelength from the header, for flats only.
+        """Get the illumination wavelength.
+
+        Get the monochromator wavelength from the header, for flats only.
         Method will need ammending if/when this value is written elsewhere
         """
         val = md.get("OBJECT").rstrip().lstrip()
         if self.translate_imgType(md) != 'flat':
             return float('nan') # defaults to NaN if not a flat
-        if val[0:3].isdigit():
+        if val[0:4].isdigit():
+            wavelength = float(val[0:4])
+        elif val[0:3].isdigit():
             wavelength = float(val[0:3])
-            if wavelength<300: #blindly using the first 3 digits, so if >=1000nm could be bad
+            if wavelength<300 or wavelength>1150: #We don't know what might be stored here,
+                                                  #so a little sanity checking is good
```

Comment:
Looks good, this is certainly more readable than the code I suggested. It might be good to define `MIN_WAVELENGTH=300` and `MAX_WAVELENGTH=1150` at the beginning of the code in case these numbers ever need to change and/or are used anywhere else in the code.